Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -869,6 +869,97 @@ final class DeclarationTests: XCTestCase {
     )
   }
 
+  func testInitAccessor() {
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init {}
+          get {}
+          set {}
+        }
+      }
+      """
+    )
+
+    assertParse(
+      """
+      struct S {
+        let _value: Int
+
+        init() {
+        }
+      }
+      """
+    )
+
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init(newValue) {}
+          get {}
+          set(newValue) {}
+        }
+      }
+      """
+    )
+
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init(newValue) {}
+          get {}
+          set(newValue) {}
+        }
+      }
+      """
+    )
+  }
+
+  func testInitAccessorEffects() {
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init initializes(x) {}
+        }
+      }
+      """
+    )
+
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init accesses(x) {}
+        }
+      }
+      """
+    )
+
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init initializes(x) accesses(y) {}
+        }
+      }
+      """
+    )
+
+    assertParse(
+      """
+      struct S {
+        var value: Int {
+          init initializes(x, y, z) accesses(x, y, z) {}
+        }
+      }
+      """
+    )
```

Comment:
Could you add a test case for the following? We should be offering a Fix-It to switch the positions of `initializes(x)` and `accesses(y)`

```swift
struct S {
  var value: Int {
    init accesses(y) initializes(x) {}
  }
}
```

--- 

Also I think the following tests would be useful

```swift
struct S {
  var value: Int {
    get accesses(y) {}
  }
}
```

```swift
struct S {
  var value: Int {
    init throws {}
  }
}
```